src/init/starship.ps1

```diff
@@ -177,6 +192,10 @@ $null = New-Module starship {
 
     }
 
+    # The following regex is used to strip out ANSI sequences from a string,
+    # giving us just plain text. This is useful for string length calculation.
+    $ansiRegex = '([\u001B\u009B][[\]()#;?]*(?:(?:(?:[a-zA-Z\d]*(?:;[-a-zA-Z\d\/#&.:=?%@~_]*)*)?\u0007)|(?:(?:\d{1,4}(?:;\d{0,4})*)?[\dA-PR-TZcf-ntqry=><~])))'
```

Comment:
Is there a way to make this simpler? In practice, starship doesn't use all that many escapes. In shells that require it, only the `\x1b...m` escape is marked as zero-length.

Reply:
Starship might not use them internally, but in theory users can specify anything in the starship.toml file, right? The regex is taken from https://github.com/chalk/ansi-regex/blob/main/index.js. I included it as-is for the sake of completeness. I don't suppose there'll be noticeable penalty for searching all the characters?

Reply:
I'll try testing that, too. It might also be worth to compile this regex.

Reply:
```suggestion
    $ansiRegex = [regex]'([\u001B\u009B][[\]()#;?]*(?:(?:(?:[a-zA-Z\d]*(?:;[-a-zA-Z\d\/#&.:=?%@~_]*)*)?\u0007)|(?:(?:\d{1,4}(?:;\d{0,4})*)?[\dA-PR-TZcf-ntqry=><~])))'
```